### PR TITLE
feat: status conditions

### PR DESCRIPTION
**Pull request by moolen**
This includes/requires #24 & #23

* adds a basic controller test
* sets ES `condition=Ready` and refreshTime
* ~~adds a noop provider (for e2e testing)~~

I'll rebase once the above mentioned PRs are merged.

### CODE REVIEW

File: pkg/controllers/externalsecret/externalsecret_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+			}, timeout, interval).Should(BeTrue())
+
+		})
+	})
+})
+
+// CreateNamespace creates a new namespace in the cluster.
+func CreateNamespace(baseName string, c client.Client) (string, error) {
+	ts := time.Now().UnixNano()
+	genName := fmt.Sprintf("ctrl-test-%v-%v-", baseName, ts)
```
Comment (mcavoyk) [inline pkg/controllers/externalsecret/externalsecret_controller_test.go:166]: Is there a need to append the current timestamp to the name? `GenerateName` should guarantee a name is used that is unique and does not conflict.

File: pkg/controllers/externalsecret/suite_test.go [hunk 2]
```diff
@@ -63,15 +64,28 @@ var _ = BeforeSuite(func(done Done) {
...
-
 	// +kubebuilder:scaffold:scheme
 
-	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: scheme.Scheme,
+	})
 	Expect(err).ToNot(HaveOccurred())
+
+	k8sClient = k8sManager.GetClient()
```
Comment (mcavoyk) [inline pkg/controllers/externalsecret/suite_test.go:74]:
```suggestion
	// do not use k8sManager.GetClient()
	// see https://github.com/kubernetes-sigs/controller-runtime/issues/343#issuecomment-469435686
	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
```

**Review by mcavoyk — APPROVED** [review]
Looks good, just a couple minor suggestions 👍

File: pkg/controllers/externalsecret/suite_test.go (hunk 2 above)
Comment (moolen) [reply pkg/controllers/externalsecret/suite_test.go]: thanks a lot, i remember debugging this in the past. The kubebuilder example project & book explicitly says to use the manager's client in favor of the custom client :disappointed: 

https://github.com/kubernetes-sigs/kubebuilder/blame/master/docs/book/src/cronjob-tutorial/testdata/project/controllers/suite_test.go#L115-L116

https://book.kubebuilder.io/cronjob-tutorial/writing-tests.html

Comment (mcavoyk) [reply pkg/controllers/externalsecret/suite_test.go]: Yeah, those kubebuilder docs may need to be updated

**Comment by mcavoyk** [comment]
Is this good to merge @moolen?

**Comment by moolen** [comment]
Yes it is! I just opened up https://github.com/kubernetes-sigs/kubebuilder/issues/2018 to document that issue in kubebuilder

**Pull request merged**